demo/full/scripts/controllers/Settings.jsx

```diff
@@ -0,0 +1,259 @@
+    this.setState({ offlineRetry: evt.target.value });
+
+  onEnableFastSwitchingClick = (evt) =>
+    this.setState({ enableFastSwitching: getCheckBoxValue(evt.target) });
+
+  onAudioTrackSwitchingModeChange = (evt) =>
+    this.setState({ audioTrackSwitchingMode: evt.target.value });
+
+  onCodecSwitchChange = (evt) =>
+    this.setState({ onCodecSwitch: evt.target.value });
+
+  onWantedBufferAheadInput = (evt) =>
+    this.setState({ wantedBufferAhead: evt.target.value });
+
+  onMaxBufferBehindInput = (evt) =>
+    this.setState({ maxBufferBehind: evt.target.value });
+
+  onMaxBufferAheadInput = (evt) => 
+    this.setState({ maxBufferAhead: evt.target.value });
+    
+  render() {
+    const {
+      autoPlay,
+      manualBrSwitchingMode,
+      initialVideoBr,
+      initialAudioBr,
+      minVideoBr,
+      minAudioBr,
+      maxVideoBr,
+      maxAudioBr,
+      limitVideoWidth,
+      throttleVideoBitrateWhenHidden,
+      stopAtEnd,
+      segmentRetry,
+      manifestRetry,
+      offlineRetry,
+      enableFastSwitching,
+      audioTrackSwitchingMode,
+      onCodecSwitch,
+      wantedBufferAhead,
+      maxBufferAhead,
+      maxBufferBehind,
+    } = this.state;
+
+    const initialSettings = {
+      initialVideoBr,
+      initialAudioBr,
+      minAudioBr,
+      minVideoBr,
+      maxVideoBr,
+      maxAudioBr,
+      limitVideoWidth,
+      throttleVideoBitrateWhenHidden,
+      stopAtEnd,
+      onInitialVideoBrInput: this.onInitialVideoBrInput,
+      onInitialAudioBrInput: this.onInitialAudioBrInput,
+      onMinAudioBrInput: this.onMinAudioBrInput,
+      onMinVideoBrInput: this.onMinVideoBrInput,
+      onMaxAudioBrInput: this.onMaxAudioBrInput,
+      onMaxVideoBrInput: this.onMaxVideoBrInput,
+      onLimitVideoWidthClick: this.onLimitVideoWidthClick,
+      onThrottleVideoBitrateWhenHiddenClick:
+        this.onThrottleVideoBitrateWhenHiddenClick,
+      onStopAtEndClick: this.onStopAtEndClick,
+    };
+
+    const networkConfig = {
+      segmentRetry,
+      manifestRetry,
+      offlineRetry,
+      onSegmentRetryInput: this.onSegmentRetryInput,
+      onManifestRetryInput: this.onManifestRetryInput,
+      onOfflineRetryInput: this.onOfflineRetryInput,
+    };
+
+    const trackSwitchModeConfig = {
+      enableFastSwitching,
+      audioTrackSwitchingMode,
+      onCodecSwitch,
+      onEnableFastSwitchingClick: this.onEnableFastSwitchingClick,
+      onAudioTrackSwitchingModeChange: this.onAudioTrackSwitchingModeChange,
+      onCodecSwitchChange: this.onCodecSwitchChange,
+    };
+
+    const buffersConfig = {
+      wantedBufferAhead,
+      maxBufferAhead,
+      maxBufferBehind,
+      onWantedBufferAheadInput: this.onWantedBufferAheadInput,
+      onMaxBufferAheadInput: this.onMaxBufferAheadInput,
+      onMaxBufferBehindInput: this.onMaxBufferBehindInput,
+    };
+
+    if (!this.props.showOptions) {
+      return null;
+    }
+
+    return (
+      <div style={{ marginTop: 5, display: "flex", }}>
```

Comment:
We generally put the style in an external CSS file (`demo/full/styles/style.css`).
What are the advantages of putting it inline like that?

Reply:
yes, I forgot to move that code inside the file u mentioned, was useful when developing but not anymore.